### PR TITLE
8290531: Loom: Parallelize a few tests more deeply

### PR DESCRIPTION
**Pull request by shipilev**
Some of the newly added Loom tests are long-running, and thus they delay the completion of otherwise very parallel tier1/jdk_loom. We can parallelize them a bit better to avoid these testing stalls. 

Improvements on Linux x86_64, TR 3970X, `jdk_loom hotspot_loom`:

```
# release before
real 4m41.424s
user 24m18.064s
sys 1m16.440s

# release after
real 2m47.769s   ; -40%
user 23m44.622s
sys 1m15.240s


# fastdebug before
real 5m38.078s
user 67m23.516s
sys 1m56.446s

# fastdebug after
real 4m9.249s    ; -26%
user 67m21.940s
sys 1m57.625s

```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290531](https://bugs.openjdk.org/browse/JDK-8290531): Loom: Parallelize a few tests more deeply

### Reviewers
 * [Ron Pressler](https://openjdk.org/census#rpressler) (@pron - Committer) ⚠️ Review applies to [3f9bf337](https://git.openjdk.org/jdk/pull/9554/files/3f9bf3377d9744a596eeb47ffcd93ab68a74747a)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [3f9bf337](https://git.openjdk.org/jdk/pull/9554/files/3f9bf3377d9744a596eeb47ffcd93ab68a74747a)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9554/head:pull/9554` \
`$ git checkout pull/9554`

Update a local copy of the PR: \
`$ git checkout pull/9554` \
`$ git pull https://git.openjdk.org/jdk pull/9554/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9554`

View PR using the GUI difftool: \
`$ git pr show -t 9554`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9554.diff">https://git.openjdk.org/jdk/pull/9554.diff</a>

</details>

### CODE REVIEW

**Comment by shipilev** [comment]
Any comments, @pron, @AlanBateman?

**Review by pron — APPROVED** [review]
The changes to Fuzz.java seem fine to me.

**Comment by shipilev** [comment]
Is @AlanBateman on vacation? Can I have someone review the NIO test change? Thanks!

**Comment by pron** [comment]
He is. He'll be back next week, but maybe someone else wants to review this.

File: test/jdk/java/nio/channels/vthread/BlockingChannelOps.java [hunk 1]
```diff
@@ -22,17 +22,25 @@
...
  * @summary Basic tests of virtual threads doing blocking I/O with NIO channels
  * @enablePreview
  * @library /test/lib
  * @run testng/othervm/timeout=300 BlockingChannelOps
+ */
+
+/**
+ * @test id=useDirectRegister
+ * @bug 8284161
+ * @summary Basic tests of virtual threads doing blocking I/O with NIO channels
```
Comment (jaikiran) [inline test/jdk/java/nio/channels/vthread/BlockingChannelOps.java:36]: Hello Aleksey, I see that previously the `@bug` and `@summary` was only there on one test definition. So maybe remove this duplicated `@bug` and `@summary` from this new test definition?
Other than that, this test change looks OK to me.

Comment (jaikiran) [reply test/jdk/java/nio/channels/vthread/BlockingChannelOps.java:36]: On a related note - I had a brief look at the jtreg docs including the FAQ https://openjdk.org/jtreg/faq.html and I don't see any mention about how it treats multiple test definitions in a single source file, when it comes to running them concurrently. But I believe you have more experience with it and have already tested the difference in parallelism, so I think what you suggest here is fine.

Comment (shipilev) [reply test/jdk/java/nio/channels/vthread/BlockingChannelOps.java:36]: Yes, this is a usual thing to replicate the test sections. It is also a usual practice to copy-paste the entirety of run definition/config.

**Review by jaikiran — APPROVED** [review]

**Comment by shipilev** [comment]
Hey @AlanBateman, welcome back? Please see if you agree with network tests change?

**Comment by AlanBateman** [comment]
No objection to changing these tests. We should probably do test/jdk/java/net/vthread/BlockingSocketOps.java at the same time as it also runs the sockets tests with the Poller in both configurations.

For the test ids then I'd prefer not use two styles (id=useDirectRegister and id=no-continuations). Can you change these to `indirect-register` and `no-vmcontinuations`?

I'd prefer if the `@bug` tag is not copied, only because we add to the bug list when extending tests.

**Comment by shipilev** [comment]
> No objection to changing these tests.

Did the changes in new commit.

**Review by AlanBateman — APPROVED** [review]

**Comment by shipilev** [comment]
Thank you!

/integrate

**Pull request closed without merging**